src/lager_config.erl

```diff
@@ -67,21 +60,76 @@ get(Key) ->
     get({?DEFAULT_SINK, Key}, undefined).
 
 get({Sink, Key}, Default) ->
-    try
-    case ets:lookup(?TBL, {Sink, Key}) of
-        [] ->
-            Default;
-        [{{Sink, Key}, Res}] ->
-            Res
-    end
-    catch
-        _:_ ->
-            Default
-    end;
+    lookup({Sink, Key}, Default);
 get(Key, Default) ->
     get({?DEFAULT_SINK, Key}, Default).
 
 set({Sink, Key}, Value) ->
-    ets:insert(?TBL, {{Sink, Key}, Value});
+    insert({Sink, Key}, Value);
 set(Key, Value) ->
     set({?DEFAULT_SINK, Key}, Value).
+
+%% check if we can use persistent_term for config
+%% persistent term was added in OTP 21.2 but we can't
+%% check minor versions with macros so we're stuck waiting
+%% for OTP 22
+-if(?OTP_RELEASE < 22).
```

Comment:
This preprocessor syntax sadly only works in >= 21, the compiler for OTP 20 and 19 barf on it.